dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerTest.java

```diff
@@ -1152,22 +1151,39 @@ public void testFindLinkageProblems_referenceToJava11Method() throws IOException
 
   @Test
   public void testFindLinkageProblems_unusedClassReferenceInByteCode() throws IOException {
-    // JDK's tools.jar contains com.sun.tools.internal.ws.wscompile.WsgenOptions class. The class
-    // has the class reference to JDK's com.sun.xml.internal.ws.api.BindingID$SOAPHTTPImpl in its
-    // constant pool section. The referenced class is private but it's not used in the JVM
-    // instructions in the class file.
-    // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1608
-
-    String javaHomeDirectory = System.getProperty("java.home");
-    Path toolsJar = Paths.get(javaHomeDirectory, "..", "lib", "tools.jar");
-    Artifact toolsArtifact =
-        new DefaultArtifact("com.sun:tools:jar:1.8").setFile(toolsJar.toFile());
-
-    LinkageChecker linkageChecker =
-        LinkageChecker.create(ImmutableList.of(new ClassPathEntry(toolsArtifact)));
+    // com.sun.tools.ws.wscompile.WsgenOptions in jaxws-tools has the class reference to
+    // com.sun.xml.ws.api.BindingID$SOAPHTTPImpl in its constant pool section. The referenced class
+    // is private but it's not used in the JVM instruction in the referencing class file. Therefore
+    // the Linkage Checker should not report it as a symbol problem.
+    // The problem was observed in Java 8's tools.jar (
+    // https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1608). In Java 11, the
+    // JDK does not provide the class any more. The two artifacts below have the classes with
+    // different packages (they are no longer 'internal').
+    ImmutableList<ClassPathEntry> classPath =
+        TestHelper.resolve(
+            "com.sun.xml.ws:jaxws-tools:2.3.3", // This contains WsgenOptions
+            "com.sun.xml.ws:jaxws-rt:2.3.3"); // This contains BindingID$SOAPHTTPImpl
+    String wsgenOptionsClassName = "com.sun.tools.ws.wscompile.WsgenOptions";
+
+    // Ensure the class path contains the two classes in the issue (#1608).
+    ClassDumper classDumper = ClassDumper.create(classPath);
+    SymbolReferences symbolReferences = classDumper.findSymbolReferences();
+    ImmutableSet<ClassSymbol> classSymbols =
+        symbolReferences.getClassSymbols(new ClassFile(classPath.get(0), wsgenOptionsClassName));
+    Truth.assertThat(classSymbols)
+        .contains(new ClassSymbol("com.sun.xml.ws.api.BindingID$SOAPHTTPImpl"));
```

Comment:
This class is a private class.

![Uploading Screen Shot 2020-11-19 at 15.10.31.png…]()